From 51508ed6e90b5b4855914bec4adadf2925627cd9 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Fri, 28 Aug 2026 10:26:07 +0200
Subject: [PATCH 30/52] UPSTREAM: iconvdata: SHIFT_JISX0213 decoding lacks
 pending character reset (CVE-2026-77117)

This fixes bug 34556.

Reviewed-by: Carlos O'Donell <carlos@redhat.com>
(cherry picked from commit 68d94bbe50b7577d48998107d632ef3a0df050e3)
---
 iconvdata/shift_jisx0213.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/iconvdata/shift_jisx0213.c b/iconvdata/shift_jisx0213.c
index 61c9c3ce6d..e9179f605e 100644
--- a/iconvdata/shift_jisx0213.c
+++ b/iconvdata/shift_jisx0213.c
@@ -226,6 +226,9 @@
 	    STANDARD_FROM_LOOP_ERR_HANDLER (1);				      \
 	  }								      \
       }									      \
+    else 								      \
+      /* There was a pending character.  Clear it.  */			      \
+      *statep = 0;							      \
 									      \
     put32 (outptr, ch);							      \
     outptr += 4;							      \
-- 
2.55.0

